src/suites/cts/robust_access.spec.ts

```diff
@@ -0,0 +1,364 @@
+export const description = `Tests to check array clamping in shaders is correctly implemented including vector / matrix indexing`;
+
+import { TestGroup, pcombine, pfilter, poptions } from '../../framework/index.js';
+
+import { GPUTest } from './gpu_test.js';
+
+export const g = new TestGroup(GPUTest);
+
+// Utilities that should probably live in some shared place.
+function copyArrayBuffer(src: ArrayBuffer): ArrayBuffer {
+  const dst = new ArrayBuffer(src.byteLength);
+  new Uint8Array(dst).set(new Uint8Array(src));
+  return dst;
+}
+
+const kUintMax = 4294967295;
+const kIntMax = 2147483647;
+
+// A small utility to test shaders:
+//  - it wraps the source into a small harness that checks the runTest() function returns 0.
+//  - it runs the shader with the testBindings set as bindgroup 0.
+//
+// The shader also has access to a uniform value that's equal to 0u to avoid constant propagation
+// in the shader compiler.
+async function runShaderTest(
+  t: GPUTest,
+  stage: GPUShaderStageFlags,
+  testSource: string,
+  testBindings: sequence<GPUBindGroupBinding>
+): Promise<void> {
+  if (stage !== GPUShaderStage.COMPUTE) {
+    throw new Error('Only know how to deal with compute for now');
+  }
+  const [constantsBuffer, constantsInit] = t.device.createBufferMapped({
+    size: 4,
+    usage: GPUBufferUsage.COPY_DST | GPUBufferUsage.UNIFORM,
+  });
+
+  const constantsData = new Uint32Array(constantsInit);
+  constantsData[0] = 1;
+  constantsBuffer.unmap();
+
+  const resultBuffer = t.device.createBuffer({
+    size: 4,
+    usage: GPUBufferUsage.COPY_SRC | GPUBufferUsage.STORAGE,
+  });
+
+  const source =
+    `#version 450
+    layout(std140, set = 1, binding = 0) uniform Constants {
+      uint one;
+    };
+    layout(std430, set = 1, binding = 1) buffer Result {
+      uint result;
+    };
+    ` +
+    testSource +
+    `
+    void main() {
+      result = runTest();
+    }`;
+
+  const pipeline = t.device.createComputePipeline({
+    computeStage: {
+      entryPoint: 'main',
+      module: t.makeShaderModuleFromGLSL('compute', source),
+    },
+  });
+
+  const group = t.device.createBindGroup({
+    layout: pipeline.getBindGroupLayout(1),
+    bindings: [
+      { binding: 0, resource: { buffer: constantsBuffer } },
+      { binding: 1, resource: { buffer: resultBuffer } },
+    ],
+  });
+
+  const testGroup = t.device.createBindGroup({
+    layout: pipeline.getBindGroupLayout(0),
+    bindings: testBindings,
+  });
+
+  const encoder = t.device.createCommandEncoder();
+  const pass = encoder.beginComputePass();
+  pass.setPipeline(pipeline);
+  pass.setBindGroup(0, testGroup);
+  pass.setBindGroup(1, group);
+  pass.dispatch(1);
+  pass.endPass();
+
+  t.queue.submit([encoder.finish()]);
+
+  await t.expectContents(resultBuffer, new Uint32Array([0]));
```

Comment:
expectContents isn't async.
```suggestion
  t.expectContents(resultBuffer, new Uint32Array([0]));
```